woocommerce/woocommerce-android
# Add "Create Test Order" feature

Part of #9459


### Description


We aim to assist new merchants in getting started by guiding them through the process of creating test orders as customers. This will also introduce them to order management features within the Woo app.

To facilitate this, we're introducing a feature on the empty order list screen for merchants who have set up their stores. This feature will direct them to their store's website in an external web browser, enabling them to place their first orders.

Once these initial orders are placed, merchants will receive notifications through the app. They can then begin to manage these orders, including fulfilling or refunding them, using the app's features.

The conditions to make this feature appear are:
- Site has been launched
- Site has published at least 1 product
- Site has set up at least 1 payment method.


### Testing instructions


1. Start with a test site with no order, at least one payment method, and make sure the site is accessible publicly already,
2. Go to Orders list in the app. Check that the app shows the "Try a test order" empty order screen like in the screenshot below,
3. Tap "Try a test order" and ensure the dialog is shown.
4. Tap the "Start test order" button and ensure it opens to the site on a browser window.

Negative case:
1. Start with a test site with no order, no payment method and/or private site,
2. Go to Orders list in the app. Make sure the default empty order screen is shown instead.

### Images/gif


 | Default empty order screen | "Try test order" Empty order screen | Try test order dialog |
|--------|--------|--------|
| ![Screenshot_20230801_160545](https://github.com/woocommerce/woocommerce-android/assets/266376/b819e780-e913-40c9-8e86-d2aeb913c6b0) | ![Screenshot_20230801_155700](https://github.com/woocommerce/woocommerce-android/assets/266376/a7c93c3e-52c0-41a1-8287-db3c3e588f0a)  | ![Screenshot_20230801_155647](https://github.com/woocommerce/woocommerce-android/assets/266376/adc9edf9-8014-4a50-9fbb-7aefca408115) |
